### ISSUE TITLE
Feature Request: define first searchable column

### ISSUE THREAD

**Issue by robbat2** [gwsw/less#44]
TL;DR:
Make it possible to declare which column any regex searches should start on, by skipping a margin/gutter on the left hand side.

This is mostly to help integration with `bat` as a formatter before `less`
https://github.com/sharkdp/bat/issues/456

`bat` includes a line number gutter, but naively this breaks regex searches that start with `^`, as you now need to include for the gutter content in your search string.

If less just started searching an column N rather than column 0, then usage would be much easier. It should be possible to pass the value of N via both command-line options AND environment variables (`bat` for example would set it in the environment when calling `less`).

**Comment by gwsw** [comment]
I think this is largely addressed by the --no-search-headers option added in bd63d7d8e2029ee3f860d35a5ab45b750abc9489. You can define a header column and make it non-searchable:
```less --header=0,6 --no-search-headers ...```

**Comment by robbat2** [comment]
Thanks; I just tested it.
I think there is a weird interaction in the `--header` calculation and ANSI codes.
If the number of ANSI codes for the left gutter are inconsistent, then the left gutter region cannot be defined by a simple offset.

Here's a sample input file to reproduce:
https://dev.gentoo.org/~robbat2/less-sample-input-for-searchable-but-ansi-problems.patch
That puts the first character of the main section at column 35 in *most* cases.
To get the search header skipping, I need to use 0,35 per that.
```< input less -RF --header=0,35 --no-search-header```
That makes search `/^commit` work correctly.
However searching for `/^STDIN` does not work, because the gutter size is different.

To make `/^STDIN` work, it would have to be `0,11`.

`bat` could fix this by emitting more ANSI in those first 3 lines, but I'm wondering if there is another fix possible, to define the left gutter-to-content-transition character by some regex, e.g. `^[^│]+│ `.

**Comment by gwsw** [comment]
You're right, this was not working if there are ANSI sequences in the header column. This should be fixed in a92036a4672b377a03436bb929447eb24881e3b0. The column number passed to the --header option should be a number of printable characters, not including any ANSI sequences, so for your example the command should be
```
less -RF --header=0,9 --no-search-header
```
This will also make horizontal scrolling work correctly.

**Comment by robbat2** [comment]
Works great; thanks! Looking forward to seeing it in the next non-beta release.
